tidy3d/components/data/monitor_data_n2f.py

```diff
@@ -1,31 +1,85 @@
 """ Monitor Level Data associated with near-to-far transformation monitors."""
 from __future__ import annotations
+from abc import ABC, abstractmethod
 from typing import Dict, Union, Callable, Tuple
 
 import xarray as xr
 import numpy as np
 import pydantic as pd
 
+from .monitor_data import MonitorData
+from .data_array import DataArray
+from .data_array import Near2FarAngleDataArray, Near2FarCartesianDataArray, Near2FarKSpaceDataArray
 from ..monitor import Near2FarAngleMonitor, Near2FarCartesianMonitor, Near2FarKSpaceMonitor
 from ..medium import Medium
+from ..types import Axis
 from ..validators import enforce_monitor_fields_present
-from ...log import SetupError
+from ...log import SetupError, log
 from ...constants import C_0, ETA_0
 
-from .monitor_data import MonitorData
-from .data_array import DataArray
-from .data_array import Near2FarAngleDataArray, Near2FarCartesianDataArray, Near2FarKSpaceDataArray
 
+RADVECTYPE = Union[Near2FarAngleDataArray, Near2FarCartesianDataArray, Near2FarKSpaceDataArray]
 
-class AbstractNear2FarData(MonitorData):
+
+class AbstractNear2FarData(MonitorData, ABC):
     """Collection of radiation vectors in the frequency domain."""
 
-    monitor: Union[Near2FarAngleMonitor, Near2FarCartesianMonitor, Near2FarKSpaceMonitor]
+    monitor: Union[Near2FarAngleMonitor, Near2FarCartesianMonitor, Near2FarKSpaceMonitor] = None
+
+    Ntheta: RADVECTYPE = pd.Field(
+        ...,
```

Comment:
Since all these quantities are required (which makes sense because the final fields etc. require all the radiation vectors), maybe I can just remove the `fields: Tuple[RadVec, ...] = pydantic.Field(["Ntheta", "Nphi", "Ltheta", "Lphi"]...` from the `AbstractNear2FarMonitor` in `monitor.py` (and make associated backend changes).

Reply:
I think so yea.